### PR TITLE
docs(kraft): introduces new section describing kraft

### PR DESCRIPTION
**Pull request by PaulRMellor**
**Documentation**

New section describing KRaft in overview, the config required and how it operates
w/ diagrams

Plus:

- Updates the intro to the configuring section
- Splits the kafka deployment procedures to KRaft and zookeeper-based
- Splits the kafka config procedures to kRaft and zookeeper-based
- Moves procedure to migrate to KRaft to sit with the KRaft overview
- Streamlines procedures and removes/moves unnecessary content following split
- Makes the KRaft and ZooKeeper sections consistent  

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [x] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

File: documentation/assemblies/deploying/assembly-kraft-mode.adoc [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+[role="_abstract"]
+KRaft (Kafka Raft metadata) mode replaces Kafka's dependency on ZooKeeper for cluster management. 
+KRaft mode simplifies the deployment and management of Kafka clusters by bringing metadata management and coordination of clusters into Kafka.
+
+Kafka in KRaft mode is designed to offer enhanced reliability, scalability, and throughput.
+Metadata operations become more efficient as they are directly integrated.
+And by removing the need to maintain a ZooKeeper cluster, there's also a reduction in the operational and security overhead.
+
+To deploy a Kafka cluster in KRaft mode, you must use `Kafka` and `KafkaNodePool` custom resources.
+The `Kafka` resource using KRaft mode must also have the annotation `strimzi.io/kraft: enabled`.
```
[the remark this answers is not in the document]
Comment (ppatierno) [reply documentation/assemblies/deploying/assembly-kraft-mode.adoc:17]: +1

File: documentation/assemblies/deploying/assembly-kraft-mode.adoc [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+* *Controller* nodes operate in the control plane to manage cluster metadata and the state of the cluster using a Raft-based consensus protocol.
+* *Broker* nodes operate in the data plane to manage the streaming of messages, receiving and storing data in topic partitions.
+* *Dual-role* nodes fulfill the responsibilities of controllers and brokers.
+
+Controllers use a metadata log, stored as a single-partition topic (`__cluster_metadata`) on every node, which records the state of the cluster. 
+An active lead controller manages updates to the metadata log, and follower controllers replicate these updates. 
+The metadata log stores information on brokers, replicas, topics, and partitions, including the state of in-sync replicas and partition leadership. 
+Kafka uses this metadata to coordinate changes and manage the cluster effectively.
+
+Brokers store the metadata log passively to stay up-to-date with the cluster's state.
```
Comment (ppatierno) [inline documentation/assemblies/deploying/assembly-kraft-mode.adoc:31]: I would also add some terminology used for broker ... "they act as observers" (as you say before that controllers are followers of the leader).

File: documentation/shared/images/kraft-dual-role-quorum.png
[the remark this answers is not in the document]
Comment (ppatierno) [reply documentation/shared/images/kraft-dual-role-quorum.png:1]: Agree with Jakub but also another comment which could make the diagram more complex.
As we mentioned before, the brokers act as "observers" so they are fetching the metadata topic as well.
From the diagram it's not clear, where we have "data replication" only across broker and "metadata replication" only across controllers while it should happen even across brokers (together with the "data replication"). Same comment applies to the next diagram.

File: documentation/shared/images/kraft-single-role-quorum.png
[the remark this answers is not in the document]
Comment (PaulRMellor) [reply documentation/shared/images/kraft-single-role-quorum.png:1]: I've updated both diagrams to include pod and kafka node

File: documentation/shared/images/kraft-dual-role-quorum.png
[the remark this answers is not in the document]
Comment (PaulRMellor) [reply documentation/shared/images/kraft-dual-role-quorum.png:1]: I've updated both diagrams. I think they better illustrate the surrounding text now. I've updated the combined node diagram without removing the broker-only nodes, but updated the surrounding text, assuming that was what we were aiming for with the second suggestion.

**Review by scholzj — APPROVED** [review]
One nit, but LGTM otherwise. Thanks.

**Comment by ppatierno** [comment]
@PaulRMellor I was wondering if, for the KRaft single quorum diagram, we should have brokers and controllers on the same level, because the metadata replication red line is going across the central controller which could confuse imho. Maybe the nodes should be put the same way as the other diagram. Wdyt?

**Comment by PaulRMellor** [comment]
> @PaulRMellor I was wondering if, for the KRaft single quorum diagram, we should have brokers and controllers on the same level, because the metadata replication red line is going across the central controller which could confuse imho. Maybe the nodes should be put the same way as the other diagram. Wdyt?

Thanks @ppatierno. I've rearranged as suggested. I did consider it, but I thought it might shrink too much to fit on the page, but it's okay. It's more consistent with the look of the other now too.

**Review by ppatierno — APPROVED** [review]
LGTM. Thanks!

**Comment by showuon** [comment]
Will take a look tomorrow. Thanks.

**Review by showuon — APPROVED** [review]
LGTM! Just one minor comment, but I think it's fine we don't change it. In KRaft, we call `active controller` for the controller leader, not `lead controller`, but I think it's still understandable. Just FYI.

Thanks.

**Comment by PaulRMellor** [comment]
> LGTM! Just one minor comment, but I think it's fine we don't change it. In KRaft, we call `active controller` for the controller leader, not `lead controller`, but I think it's still understandable. Just FYI.
> 
> Thanks.

Thanks @showuon -- I think it's worth making that change to be consistent

**Pull request merged**